controllers/user_reconciler.go

```diff
@@ -0,0 +1,297 @@
+package controllers
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	userv1 "github.com/openshift/api/user/v1"
+	"github.com/prometheus/common/log"
+	workshopv1 "github.com/stakater/workshop-operator/api/v1"
+	openshiftuser "github.com/stakater/workshop-operator/common/user"
+	"github.com/stakater/workshop-operator/common/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"os/exec"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+const (
+	HTPASSWD_SECRET_NAME            = "htpass-workshop-users"
+	HTPASSWD_SECRET_NAMESPACE_NAME  = "openshift-config"
+	USER_ROLE_BINDIN_NAMESPACE_NAME = "workshop-infra"
+	IDENTITY_NAME                   = "htpass-workshop-users"
+	USER_IDENTITY_MAPPING_NAME      = "htpass-workshop-users"
+)
+
+var userLabels = map[string]string{
+	"createdBy": "WorkshopOperator",
+}
+var UserLabelSelector = "createdBy=WorkshopOperator"
+
+func (r *WorkshopReconciler) reconcileUser(workshop *workshopv1.Workshop) (reconcile.Result, error) {
+	var createUsers []string
+	var userList []string
+	var skipUsers []string
+
+	totalUsers := workshop.Spec.UserDetails.NumberOfUsers
+	userPrefix := workshop.Spec.UserDetails.UserNamePrefix
+	password := workshop.Spec.UserDetails.DefaultPassword
+
+	for userSuffix := 1; userSuffix <= totalUsers; userSuffix++ {
+		userName := fmt.Sprint(userPrefix, userSuffix)
+		createUsers = append(createUsers, userName)
+	}
+	labelSelector, err := labels.Parse(UserLabelSelector)
+	if err != nil {
+		log.Errorf("Error %s", err)
+	}
+	ListUsers := &userv1.UserList{}
+	listOps := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	// list User
+	if err := r.List(context.TODO(), ListUsers, listOps); err != nil {
+		log.Errorf("Error %s", err)
```

Comment:
log.Errorf("Error %s", err) -> `log.Errorf("Error occured while fetching list of users, filtered by labels {%s}, %s", err)`